### PR TITLE
Support multi disks for caching hive files

### PR DESCRIPTION
**Pull request by lgbo-ustc**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support multi disks for caching hive files.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/Cache/ExternalDataSourceCache.cpp [hunk 1]
```diff
@@ -195,14 +228,22 @@ void ExternalDataSourceCache::initOnce(ContextPtr context, const String & root_d
     }
     LOG_INFO(
         log, "Initializing local cache for remote data sources. Local cache root path: {}, cache size limit: {}", root_dir_, limit_size_);
-    root_dir = root_dir_;
+    Poco::StringTokenizer tokenizer(root_dir_, ",");
```
Comment (kssenii) [inline src/Storages/Cache/ExternalDataSourceCache.cpp:231]: better use
```
splitInto<','>(root_dirs, root_dir_)
```

File: tests/integration/test_hive_query/configs/config.xml [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
     <local_cache_for_remote_fs>
         <enable>true</enable>
-        <root_dir>/tmp/clickhouse_local_cache</root_dir>
+        <root_dir>/tmp/clickhouse_local_cache,/tmp/clickhouse_local_cache1</root_dir>
```
Comment (kssenii) [inline tests/integration/test_hive_query/configs/config.xml:16]: Is it possible to add an integration test?

Comment (lgbo-ustc) [reply tests/integration/test_hive_query/configs/config.xml:16]: Do you mean to change the path list ?
I don't find a way to restart a container in the intergration.

Comment (kssenii) [reply tests/integration/test_hive_query/configs/config.xml:16]: I mean to check in test that both paths are used, which can be done with the help of `exec_in_container` from helpers/cluster.py

**Review by kssenii — APPROVED** [review]

**Pull request merged**